Network/Curl/Easy.hs

```diff
@@ -88,6 +89,14 @@ setopt hh o = curlPrim hh $ \ r h -> unmarshallOption (easy_um r h) o
               updateCleanup r i $
                 debug ("FREE: " ++ show x) >> curl_slist_free ip
               liftM toCode $ easy_setopt_string h i (castPtr ip)
+     , u_bytestring -- :: Int -> ByteString -> IO CurlCode
+        = \ i x ->
+           do debug ("ALLOC: " ++ show x)
+              useAsCString x $ \c_x ->
+                -- Strings passed to libcurl as 'char *' arguments, are copied
+                -- by the library, so updateCleanup is not needed
```

Comment:
In that case, the debug output above is probably confusing because it (albeit intentionally) lacks the corresponding "FREE". You could replace it with "USE: " to make that distinction.

The docs also say

> Before version 7.17.0, strings were not copied. Instead the user was forced keep them available until libcurl no longer needed them. 

But I assume we are using a more recent version?

Still, if we have access to the version in the code, it might be a good idea to add an assertion somewhere that we're only using a supported version.

Reply:
Done :+1: